entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt

```diff
@@ -581,6 +581,97 @@ class Neo4jRepository(
+                RETURN DISTINCT(labels(a))+collect(DISTINCT(a.name)) as attribute,
+                count(entity.location) as entityWithLocationCount
+            """.trimIndent()
+
+        val result = neo4jClient.query(query).fetch().all()
+        return result.map { rowResult ->
+            val entityWithLocationCount = (rowResult["entityWithLocationCount"] as Long).toInt()
+            var attributeList = (rowResult["attribute"] as List<String>)
+                .filter { it !in listOf("Attribute", "Relationship", "Property") }.toList()
+            if (entityWithLocationCount > 0) { attributeList += NGSILD_LOCATION_PROPERTY }
+            attributeList
+        }.flatten().sortedBy { it }
+    }
+
+    fun getAttributeDetails(): List<Map<String, Any>> {
+        val query =
+            """
+                OPTIONAL MATCH (prop:Attribute)<-[:HAS_VALUE]-(entity:Entity)
+                OPTIONAL MATCH (rel:Attribute)<-[:HAS_OBJECT]-(entity:Entity)
+                RETURN DISTINCT (labels(rel)) as relation, 
+                labels(entity) as typeNames,
+                collect(prop.name) as property,
+                count(entity.location) as entityWithLocationCount
+            """.trimIndent()
+
+        val result = neo4jClient.query(query).fetch().all()
+        return result.map { rowResult ->
+            val property = (rowResult["property"] as List<String>)
+            val relation = (rowResult["relation"] as List<String>)
+            var attribute: List<String> = property.plus(relation)
+                .filter { it !in listOf("Attribute", "Relationship", "Property") }.toList()
+            val entityWithLocationCount = (rowResult["entityWithLocationCount"] as Long).toInt()
+            if (entityWithLocationCount > 0) {
+                val location: List<String> = listOf(NGSILD_LOCATION_PROPERTY)
+                attribute = property.plus(location).plus(relation)
+                    .filter { it !in listOf("Attribute", "Relationship", "Property") }.toList()
+            }
+            val attributeName = attribute.sortedBy { it }
+            attributeName.map { attributeName ->
+                mapOf(
+                    "attribute" to attributeName,
+                    "typeNames" to (rowResult["typeNames"] as List<Any>)
+                        .filter { !authorizationEntitiesTypes.plus("Entity").contains(it) }.toSet(),
+                )
+            }
+        }.flatten()
+    }
+    fun getAttributeInformation(expandedType: String): Map<String, Any> {
+        val relationshipQuery =
+            """
+                MATCH (a:Attribute:`$expandedType`)
+                OPTIONAL MATCH (a)<-[:HAS_OBJECT|:HAS_VALUE]-(entity:Entity)
+                return DISTINCT (labels(a)) as attributeTypes,
+                labels(entity) as typeNames,
+                count(a) as attributeCount,
+                count(entity.location) as entityLocationCount
```

Comment:
I don't get why you're returning location here.

Reply:
location is geoProperty and also an attribute, so I am querying location.